### PR TITLE
Feature/latency budget qos policy [6064]

### PR DESCRIPTION
**Pull request by laura-eprosima**

### CODE REVIEW

File: Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (LuisGP) [inline Testing/Temporary/CTestCostData.txt]: Remove this file.

File: test/blackbox/BlackboxTestsLatencyBudgetQos.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#include "BlackboxTests.hpp"
+
+#include "LatencyBudgetReqRepHelloWorldRequester.hpp"
+
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+TEST(LatencyBudgetQos, DurationCheck)
+{
+
```
Comment (LuisGP) [inline test/blackbox/BlackboxTestsLatencyBudgetQos.cpp]: Remove this empty line.

File: test/blackbox/LatencyBudgetReqRepHelloWorldRequester.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (LuisGP) [inline test/blackbox/LatencyBudgetReqRepHelloWorldRequester.cpp]:
```suggestion
// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (LuisGP) [inline test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp]: Can this file be merged with `ReqRepHelloWorldRequester` adding getters and setters, and functionality for latency purposes?

```suggestion
// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/blackbox/PubSubReader.hpp [hunk 5]
```diff
@@ -33,7 +33,6 @@
 #include <fastrtps/xmlparser/XMLTree.h>
 #include <fastrtps/utils/IPLocator.h>
 #include <fastrtps/transport/UDPv4TransportDescriptor.h>
-
```
Comment (LuisGP) [inline test/blackbox/PubSubReader.hpp]: This blank line separates logical #include units. Why did you remove it?

File: Testing/Temporary/CTestCostData.txt (hunk 1 above)
Comment (laura-eprosima) [reply Testing/Temporary/CTestCostData.txt]: Done

File: test/blackbox/BlackboxTestsLatencyBudgetQos.cpp (hunk 2 above)
Comment (laura-eprosima) [reply test/blackbox/BlackboxTestsLatencyBudgetQos.cpp]: Done

File: test/blackbox/LatencyBudgetReqRepHelloWorldRequester.cpp (hunk 3 above)
Comment (laura-eprosima) [reply test/blackbox/LatencyBudgetReqRepHelloWorldRequester.cpp]: Done

File: test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp (hunk 4 above)
Comment (laura-eprosima) [reply test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp]: Done

File: test/blackbox/PubSubReader.hpp (hunk 5 above)
Comment (laura-eprosima) [reply test/blackbox/PubSubReader.hpp]: Changed

File: test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp (hunk 4 above)
Comment (laura-eprosima) [reply test/blackbox/LatencyBudgetReqRepHelloWorldRequester.hpp]: Changed

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/262/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/238/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/261/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/307/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/282/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/305/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/530/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/472/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/526/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/568/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/495/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/564/)

File: test/blackbox/PubSubReader.hpp [hunk 6]
```diff
@@ -483,6 +483,17 @@ class PubSubReader
         return *this;
     }
 
+    PubSubReader& latency_budget_duration(const eprosima::fastrtps::Duration_t latency_duration)
```
Comment (LuisGP) [inline test/blackbox/PubSubReader.hpp]:
```suggestion
    PubSubReader& latency_budget_duration(
            const eprosima::fastrtps::Duration_t& latency_duration)
```

File: test/blackbox/PubSubWriter.hpp [hunk 7]
```diff
@@ -480,6 +480,17 @@ class PubSubWriter
         return *this;
     }
 
+    PubSubWriter& latency_budget_duration(const eprosima::fastrtps::Duration_t latency_duration)
```
Comment (LuisGP) [inline test/blackbox/PubSubWriter.hpp]:
```suggestion
    PubSubWriter& latency_budget_duration(
            const eprosima::fastrtps::Duration_t& latency_duration)
```

File: test/blackbox/ReqRepHelloWorldRequester.cpp [hunk 8]
```diff
@@ -76,6 +76,34 @@ void ReqRepHelloWorldRequester::init()
     initialized_ = true;
 }
 
+void ReqRepHelloWorldRequester::init_with_latency(
+        eprosima::fastrtps::Duration_t latency_budget_duration_pub,
+        eprosima::fastrtps::Duration_t latency_budget_duration_sub)
```
Comment (LuisGP) [inline test/blackbox/ReqRepHelloWorldRequester.cpp]: Change arguments to const reference.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/585/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/509/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/581/)

File: test/blackbox/PubSubWriter.hpp (hunk 7 above)
Comment (LuisGP) [reply test/blackbox/PubSubWriter.hpp]: You missed the reference.

File: test/blackbox/PubSubReader.hpp (hunk 6 above)
Comment (LuisGP) [reply test/blackbox/PubSubReader.hpp]: You missed the reference.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/588/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/511/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/584/)

**Comment by raquelalvarezbanos** [comment]
Unrelated Windows failure
